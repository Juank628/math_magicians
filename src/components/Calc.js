import React, { useState } from 'react';
import calculate from '../logic/calculate';
import Button from './Button';
import styles from './Calc.module.css';

export default function Calc() {
  const [displayValue, setDisplayValue] = useState(['0']);
  const [calculatorData, setCalculatorData] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const updateDisplay = (newCalculatorData) => {
    const { next, operation, total } = newCalculatorData;
    if (next === null && operation === null && total === null) {
      setDisplayValue('0');
      return;
    }
    if (next === null) {
      setDisplayValue(total);
      return;
    }
    setDisplayValue(next);
  };

  const btnClickHandler = (e) => {
    const button = e.target.textContent;
    const newCalculatorData = calculate(calculatorData, button);
    updateDisplay(newCalculatorData);
    setCalculatorData(newCalculatorData);
  };

  return (
    <section className={styles.calcContainer}>
      <input
        type="text"
        className={styles.calcDisplay}
        value={displayValue}
        disabled
      />

      <Button text="AC" clickHandler={btnClickHandler} />
      <Button text="+/-" clickHandler={btnClickHandler} />
      <Button text="%" clickHandler={btnClickHandler} />
      <Button text="รท" clickHandler={btnClickHandler} color="Orange" />

      <Button text="7" clickHandler={btnClickHandler} />
      <Button text="8" clickHandler={btnClickHandler} />
      <Button text="9" clickHandler={btnClickHandler} />
      <Button text="x" clickHandler={btnClickHandler} color="Orange" />

      <Button text="4" clickHandler={btnClickHandler} />
      <Button text="5" clickHandler={btnClickHandler} />
      <Button text="6" clickHandler={btnClickHandler} />
      <Button text="-" clickHandler={btnClickHandler} color="Orange" />

      <Button text="1" clickHandler={btnClickHandler} />
      <Button text="2" clickHandler={btnClickHandler} />
      <Button text="3" clickHandler={btnClickHandler} />
      <Button text="+" clickHandler={btnClickHandler} color="Orange" />

      <Button text="0" clickHandler={btnClickHandler} columns="2" />
      <Button text="." clickHandler={btnClickHandler} />
      <Button text="=" clickHandler={btnClickHandler} color="Orange" />
    </section>
  );
}
