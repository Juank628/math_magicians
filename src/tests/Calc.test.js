import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Calc from '../components/Calc';

describe('calc tests', () => {
  it('snapshot test', () => {
    const tree = renderer.create(<Calc />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('render test', () => {
    const component = render(<Calc />);
    console.log(component);
  });
});
