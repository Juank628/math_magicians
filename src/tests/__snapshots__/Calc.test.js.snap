// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calc tests snapshot test 1`] = `
<section
  className="calcContainer"
>
  <input
    className="calcDisplay"
    disabled={true}
    type="text"
    value={
      Array [
        "0",
      ]
    }
  />
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    AC
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    +/-
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    %
  </button>
  <button
    className="calcButton calcOrange"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    รท
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    7
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    8
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    9
  </button>
  <button
    className="calcButton calcOrange"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    x
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    4
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    5
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    6
  </button>
  <button
    className="calcButton calcOrange"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    -
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    1
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    2
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    3
  </button>
  <button
    className="calcButton calcOrange"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    +
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 2",
      }
    }
    type="button"
  >
    0
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    .
  </button>
  <button
    className="calcButton calcOrange"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    =
  </button>
</section>
`;

exports[`snapshot test 1`] = `
<section
  className="calcContainer"
>
  <input
    className="calcDisplay"
    disabled={true}
    type="text"
    value={
      Array [
        "0",
      ]
    }
  />
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    AC
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    +/-
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    %
  </button>
  <button
    className="calcButton calcOrange"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    รท
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    7
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    8
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    9
  </button>
  <button
    className="calcButton calcOrange"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    x
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    4
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    5
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    6
  </button>
  <button
    className="calcButton calcOrange"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    -
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    1
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    2
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    3
  </button>
  <button
    className="calcButton calcOrange"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    +
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 2",
      }
    }
    type="button"
  >
    0
  </button>
  <button
    className="calcButton calcGray"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    .
  </button>
  <button
    className="calcButton calcOrange"
    onClick={[Function]}
    style={
      Object {
        "gridColumnEnd": "span 1",
      }
    }
    type="button"
  >
    =
  </button>
</section>
`;
